python/ray/autoscaler/v2/instance_manager/reconciler.py

```diff
@@ -38,6 +42,111 @@
 logger = logging.getLogger(__name__)
 
 
+class IInstanceUpdater(ABC):
+    """
+    An interface to for making instance update.
+    """
+
+    @abstractmethod
+    def make_update(self, instance: IMInstance) -> Optional[IMInstanceUpdateEvent]:
+        """
+        Make an instance update for the instance.
+
+        Args:
+            instance: The instance to make update.
+
+        Returns:
+            The instance update event if there's an update. None otherwise.
+        """
+        raise NotImplementedError
+
+
+class TimeoutInstanceUpdater(IInstanceUpdater):
```

Comment:
Do we need a class?

Reply:
i think maybe methods/function callables are also ok.